### PR TITLE
@W-8307462@ | Global Template Updates

### PR DESCRIPTION
**Pull request by amaskiellEG**
# Description

This PR makes the following updates:
- Across all templates with Handlebars code: Remove extra whitespace in rendered HTML code via whitespace control where appropriate
- Infobar and popup templates: check for the content zone first before rendering template in apply()
- [Visibility Check-Box Removal](https://docs.google.com/document/d/1FWtwDcm66pl8Th0G7xaUCR8S3YuxgN0Nm1-ODlIco6c/edit)
- Add double quotes to string template literals like `[data-evg-campaign-id="${context.campaign}"][data-evg-experience-id="${context.experience}"]` _(without them, it sometimes results in an invalid selector error)_

## Related Issues/Tickets

GUS: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008AbDzIAK/view

#### Templates:
All of them are in the account: `training` with dataset: `amask`
- W-8307462 | Banner with Call-To-Action
- W-8307462 | Einstein Content Recommendations
- W-8307462 | Einstein Product Recommendations
- W-8307462 | Exit Intent Pop-Up
- W-8307462 | Exit Intent Pop-Up with Email Capture
- W-8307462 | Infobar With Call-To-Action
- W-8307462 | Infobar with User Attribute & Call-To-Action
- W-8307462 | Slide-In with Call-To-Action

### CODE REVIEW

File: exit-intent-popup/template.ts [hunk 1]
```diff
@@ -22,11 +22,10 @@ export class ExitIntentPopupTemplate implements CampaignTemplateComponent {
...
     header: string = "Header Text";
 
+    @subtitle("Optional rich text field")
     @richText(true)
     subheader: string = "Subheader Text";
 
-    subheaderVisibility: boolean = true;
-
     @title("CTA Text")
     ctaText: string ="Call To Action";
```
Comment (nriserEG) [inline exit-intent-popup/template.ts:30]: Could you make this minor update too?
```suggestion
    ctaText: string = "Call To Action";
```

File: exit-intent-popup/template.ts [hunk 2]
```diff
@@ -22,11 +22,10 @@ export class ExitIntentPopupTemplate implements CampaignTemplateComponent {
...
```
Comment (nriserEG) [inline exit-intent-popup/template.ts:31]: Could you make this minor update too on line 33?
`@title('CTA Destination URL')` --> `@title("CTA Destination URL")`

File: infobar-with-cta/client.js [hunk 3]
```diff
@@ -43,8 +45,8 @@
         Evergage.cashDom("body").css({ "margin-top": "0", "margin-bottom": "0" });
     }
 
-    function control() {
-
+    function control(context) {
+        return !!context.contentZone;
```
Comment (nriserEG) [inline infobar-with-cta/client.js:49]: Control function can only return a Promise or void, and can't return a boolean. Just do what you did in apply()
```suggestion
        if (!context.contentZone) return;
```

File: infobar-with-cta/template.ts [hunk 4]
```diff
@@ -12,16 +12,13 @@ export class InfobarWithCTATemplate implements CampaignTemplateComponent {
...
-    messageVisibility: boolean = true;
-
     @title('CTA Text')
+    @subtitle("Optional text field")
     ctaText: string = "Call To Action";
 
-    @title('CTA Visibility')
-    ctaVisibility: boolean = true;
-
     @title('CTA Destination URL')
```
Comment (nriserEG) [inline infobar-with-cta/template.ts:22]: Could you make this minor update too?
```suggestion
    @title("CTA Destination URL")
```

File: infobar-with-cta/template.ts [hunk 5]
```diff
@@ -12,16 +12,13 @@ export class InfobarWithCTATemplate implements CampaignTemplateComponent {
     ])
     style: StyleField = { label: "Light on Dark", className: "evg-light-on-dark" };
 
+    @subtitle("Optional text field")
     messageText: string = "Infobar Message Text";
 
-    messageVisibility: boolean = true;
-
     @title('CTA Text')
```
Comment (nriserEG) [inline infobar-with-cta/template.ts:18]: Could you make this minor update too?
```suggestion
    @title("CTA Text")
```

File: infobar-with-user-attribute-and-cta/template.ts [hunk 6]
```diff
@@ -13,23 +13,21 @@ export class InfobarWithUserAttrAndCTATemplate implements CampaignTemplateCompon
...
     @title('CTA Text')
+    @subtitle('Optional text field')
     ctaText: string = "Call To Action";
 
-    @title('CTA Visibility')
-    ctaVisibility: boolean = true;
-
     @title('CTA Destination URL')
     @subtitle("Requires full URL string including https://")
     ctaUrl: string = "https://cumulusfinserv.com/banking";
```
Comment (nriserEG) [inline infobar-with-user-attribute-and-cta/template.ts:33]: Could you replace the single quotations with double quotations? throughout this file

File: infobar-with-user-attribute-and-cta/client.js [hunk 7]
```diff
@@ -46,8 +48,8 @@
         Evergage.cashDom("body").css({ "margin-top": "0", "margin-bottom": "0" });
     }
 
-    function control() {
-
+    function control(context) {
+        return !!context.contentZone;
```
Comment (nriserEG) [inline infobar-with-user-attribute-and-cta/client.js:52]: Control function can only return a Promise or void, and can't return a boolean. Just do what you did in apply()
```suggestion
        if (!context.contentZone) return;
```

**Review by nriserEG — CHANGES_REQUESTED** [review]
Noticed that the "Infobar With User Attribute & Call-To-Action" template name renders like this in the template list screen: 
Screen Shot 2020-12-17 at 7 36 17 PM

Could you update the "&" to "and" in the .json file for now?

File: exit-intent-popup/template.ts (hunk 1 above)
Comment (amaskiellEG) [reply exit-intent-popup/template.ts:30]: Lol How'd that happen? 🙃

File: infobar-with-cta/client.js (hunk 3 above)
Comment (amaskiellEG) [reply infobar-with-cta/client.js:49]: Hmmmm where do you see that? I thought I had tested this before and it worked as expected 🤔 

I think I see what you're talking about when I check out the type definition for `control` [here](https://github.com/evergage/evergage-product/blob/e2c06a5ac7a207db6846b6bd42d5e6aa2e6e101f/analytics/excavator/src/main/resources/built-in/build/handlebars-templates/src/services/TemplateService.ts#L120), though I'm not sure that returning void would work here then?

File: infobar-with-user-attribute-and-cta/client.js (hunk 7 above)
Comment (amaskiellEG) [reply infobar-with-user-attribute-and-cta/client.js:52]: https://github.com/evergage/evergage-global-templates/pull/28#discussion_r551612079

File: infobar-with-cta/client.js (hunk 3 above)
Comment (nriserEG) [reply infobar-with-cta/client.js:49]: https://github.com/evergage/evergage-product/blob/e2c06a5ac7a207db6846b6bd42d5e6aa2e6e101f/analytics/excavator/src/main/resources/built-in/build/handlebars-templates/src/Template.ts

Comment (amaskiellEG) [reply infobar-with-cta/client.js:49]: I just tested this out and neither the current code or your suggestion prevented the control stat from being sent when a content zone isn't selected; it seems like we actually need to do something like this:
```
return new Promise(resolve => { if (!!context.contentZone) resolve(!!context.contentZone); });
```

Comment (nriserEG) [reply infobar-with-cta/client.js]: I looked at TemplateService.ts more closely, and actually, the reason why neither your original solution nor my suggested change works to block the control impression is due to the fact that void lands in the `else` statement, firing the control impression in the downstream anyway.
Screen Shot 2021-01-05 at 5 37 31 PM

I tried your new solution with the hanging Promise and that does seem to work 👍  I would simplify to the following:
`return new Promise(resolve => { if (context.contentZone) resolve(true) });`

File: infobar-with-cta/client.js [hunk 8]
```diff
@@ -43,8 +45,8 @@
         Evergage.cashDom("body").css({ "margin-top": "0", "margin-bottom": "0" });
     }
 
-    function control() {
-
+    function control(context) {
+        return new Promise(resolve => { if (!!context.contentZone) resolve(!!context.contentZone); });
```
Comment (nriserEG) [inline infobar-with-cta/client.js:49]:
```suggestion
        return new Promise(resolve => { if (context.contentZone) resolve(true) });
```

File: infobar-with-user-attribute-and-cta/client.js [hunk 9]
```diff
@@ -46,8 +48,8 @@
         Evergage.cashDom("body").css({ "margin-top": "0", "margin-bottom": "0" });
     }
 
-    function control() {
-
+    function control(context) {
+        return new Promise(resolve => { if (!!context.contentZone) resolve(!!context.contentZone); });
```
Comment (nriserEG) [inline infobar-with-user-attribute-and-cta/client.js:52]:
```suggestion
        return new Promise(resolve => { if (context.contentZone) resolve() });
```

**Review by avkieu — APPROVED** [review]
Security Approved

**Review by nriserEG — APPROVED** [review]
Approved!

Adding this note here too so we won't forget --
Since we're updating the "Infobar With User Attribute & Call-To-Action" template name, this template will be duplicated but with the new name. As such, we'll need to ask Oleg to manually remove the old template. This should be done at the same time this ticket is released. I'll coordinate with Oleg on this

**Pull request merged**